keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -954,19 +955,20 @@ class ClusterHandler(
         subnet = subnet(cloudDriverCache),
         availabilityZones = zones
       ),
-      launchConfiguration = launchConfig.run {
+      launchConfiguration =
         LaunchConfiguration(
-          imageId = imageId,
+          imageId = launchConfig?.imageId ?: launchTemplateData!!.imageId,
```

Comment:
There is a top-level `Image` field on a server group. You could look the image id that way and avoid the conditional here.

Reply:
Fixed in bf496872